contracts/managers/timelock-manager/TimeLockManager.sol

```diff
@@ -7,38 +7,71 @@ import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 contract TimeLockManager {
     using SafeERC20 for IERC20;
 
-    event Lock(address indexed sender, uint256 amount);
-    event Unlock(address indexed receiver, uint256 amount);
+    struct TokenLock {
+        address token;
+        uint256 amount;
+        uint256 createdAt;
+        uint256 duration;
+        string info;
+        bool released;
+    }
+
+    event Lock(address indexed sender, address indexed token, uint256 amount, uint256 duration);
+    event Unlock(address indexed receiver, address indexed token, uint256 id, uint256 amount);
+
+    mapping (address => TokenLock[]) public locks;
 
-    IERC20 public token;
-    uint256 public deadline;
+    function lock(address tokenAddress, uint256 amount, uint256 duration, string memory info) public {
```

Comment:
Actually, let's keep it as is - it's for MVP anyway so it will be sufficient, and we can always add more complex timing logic calculation in `blockchain-api-service`.

Reply:
okay